tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs

```diff
@@ -136,6 +163,60 @@ public async Task JsonWebTokenWithX509PublicCertSendCertificateOnBehalfOfTestAsy
                     .ConfigureAwait(false);
 
                 Assert.IsNotNull(result.AccessToken);
+
+                //Check user cache
+                Assert.AreEqual(1, app.UserTokenCacheInternal.Accessor.GetAllAccessTokens().Count());
+
+                //Clear cache
+                app.UserTokenCacheInternal.ClearMsalCache();
+            }
+        }
+
+        [TestMethod]
+        [Description("Test for client assertion with X509 public certificate using sendCertificate")]
+        public async Task JsonWebTokenWithX509PublicCertSendCertificateSilentTestAsync()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                SetupMocks(harness.HttpManager, "https://login.microsoftonline.com/my-utid/");
+                var certificate = new X509Certificate2(
+                    ResourceHelper.GetTestResourceRelativePath("valid_cert.pfx"),
+                    MsalTestConstants.DefaultPassword);
+
+                var app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                              .WithAuthority(
+                                                                  new System.Uri("https://login.microsoftonline.com/my-utid"),
+                                                                  true).WithRedirectUri(MsalTestConstants.RedirectUri)
+                                                              .WithHttpManager(harness.HttpManager)
+                                                              .WithCertificate(certificate).BuildConcrete();
+
+                _tokenCacheHelper.PopulateCacheWithOneAccessToken(app.UserTokenCacheInternal.Accessor);
+                app.UserTokenCacheInternal.Accessor.DeleteAccessToken(
+                    new MsalAccessTokenCacheKey(
+                        MsalTestConstants.ProductionPrefNetworkEnvironment,
+                        MsalTestConstants.Utid,
+                        MsalTestConstants.UserIdentifier,
+                        MsalTestConstants.ClientId,
+                        MsalTestConstants.ScopeForAnotherResourceStr));
+
+                //Check for x5c claim
+                harness.HttpManager.AddMockHandler(CreateTokenResponseHttpHandlerWithX5CValidation(false));
+
+                var result = await app
+                    .AcquireTokenSilent(
+                        new[] { "someTestScope"},
+                        new Account(MsalTestConstants.UserIdentifier, MsalTestConstants.DisplayableId, null))
+                    .WithSendX5C(true)
```

Comment:
>.WithSendX5C(true) [](start = 20, length = 18)

how do you know that this worked as expected? And do you have to use .WithForceRefresh(true) in order for .WithSendX5C to work?

Reply:
For the mock response in this test (CreateTokenResponseHttpHandlerWithX5CValidation()) it creates a special response with a callback that checks what data was actually being sent on the wire. 
```CSharp
                    var x5c = jsonToken.Header.Where(header => header.Key == "x5c").FirstOrDefault();
                    Assert.IsTrue((x5c.Key == "x5c"), "x5c should be present");
                    Assert.AreEqual(x5c.Value.ToString(), MsalTestConstants.Defaultx5cValue);
```
See line 58 for the full context.

also, developers do not need to use WithForceRefresh() for the WithSendX5C() to work. The only reason it is being used here is because we want to force the test to send another refresh token request so that we can check the header as opposed to the test simply returning the access token from the cache.